include/aspect/velocity_boundary_conditions/gplates.h

```diff
@@ -38,80 +40,47 @@ namespace aspect
           GPlatesLookup(const Tensor<1,2> &pointone,
-                        const Tensor<1,2> &pointtwo,
-                        const double interpolation_width_);
+                        const Tensor<1,2> &pointtwo);
 
           /**
-           * Outputs the GPlates module information at model start. Need to be
-           * separated from Constructor because at construction time the
-           * SimulatorAccess is not initialized and only Rank 0 should give
-           * the screen output.
+           * Outputs the GPlates module information at model start.
            */
-          template <int dim>
-          void screen_output(const Tensor<1,2> &surface_point_one,
-                             const Tensor<1,2> &surface_point_two,
-                             const ConditionalOStream &pcout) const;
-
-          /**
-           * Check whether a file named @p filename exists.
-           */
-          bool fexists(const std::string &filename) const;
+          std::string
+          screen_output(const Tensor<1,2> &surface_point_one,
+                        const Tensor<1,2> &surface_point_two) const;
 
           /**
            * Loads a gplates .gpml velocity file. Throws an exception if the
            * file does not exist.
            */
-          void load_file(const std::string &filename,
-                         const ConditionalOStream &pcout);
+          void load_file(const std::string &filename);
 
           /**
            * Returns the computed surface velocity in cartesian coordinates.
-           * Takes as input the position and current time weight.
+           * Takes as input the position. Actual velocity interpolation is
+           * performed in spherical coordinates.
            *
            * @param position The current position to compute velocity
-           * @param time_weight A weighting between the two current timesteps
-           * n and n+1
            */
-          template <int dim>
-          Tensor<1,dim> surface_velocity(const Point<dim> &position,
-                                         const double time_weight) const;
+          Tensor<1,dim> surface_velocity(const Point<dim> &position) const;
 
         private:
-
-          /**
-           * Tables which contain the velocities
-           */
-          Table<2,Tensor<1,3> > velocity_vals;
-          Table<2,Tensor<1,3> > old_velocity_vals;
-
-          /**
-           * Table for the data point positions.
-           */
-          Table<2,Tensor<1,3> > velocity_positions;
-
           /**
-           * Pointers to the actual tables. Used to avoid unnecessary copying
-           * of values. These pointers point to either velocity_vals or
-           * old_velocity_vals.
+           * Interpolation functions to access the velocities.
            */
-          Table<2,Tensor<1,3> > *velocity_values;
-          Table<2,Tensor<1,3> > *old_velocity_values;
+          std::vector<Functions::InterpolatedUniformGridData<2> *> velocities;
```

Comment:
Can you make this a vector of `std::shared_ptr` or `std::unique_ptr`? This way you won't have to deal with manually managing memory.